jsapp/js/components/submissions/submissionUtils.mocks.ts

```diff
@@ -1,65 +1,129 @@
-export const simpleSurvey = [
+import {
+  AssetTypeName,
+  GroupTypeBeginName,
+  GroupTypeEndName,
+  MetaQuestionTypeName,
+  MiscRowTypeName,
+  QuestionTypeName,
+} from '#/constants'
+import type { AssetResponse, SubmissionResponse, SurveyChoice, SurveyRow } from '#/dataInterface'
+import { type DisplayGroup, DisplayGroupTypeName } from './submissionUtils'
+
+export const simpleSurvey: SurveyRow[] = [
   {
     name: 'start',
-    type: 'start',
+    type: MetaQuestionTypeName.start,
     $kuid: '9PMXyB7Sv',
     $autoname: 'start',
   },
   {
     name: 'end',
-    type: 'end',
+    type: MetaQuestionTypeName.end,
     $kuid: 'AkKyOYSIP',
     $autoname: 'end',
   },
   {
-    type: 'text',
+    type: QuestionTypeName.text,
     $kuid: 'uw4if17',
     label: ['First name', 'Pierwsze imię'],
     required: false,
     $autoname: 'First_name',
   },
   {
     name: 'group_favourites',
-    type: 'begin_group',
+    type: GroupTypeBeginName.begin_group,
     $kuid: 'fx8qb06',
     label: ['Favourites', 'Ulubione'],
     $autoname: 'group_favourites',
   },
   {
-    type: 'select_one',
+    type: QuestionTypeName.select_one,
     $kuid: 'az1fc41',
     label: ['Favourite color', 'Ulubiony kolor'],
     required: false,
     $autoname: 'Favourite_color',
     select_from_list_name: 'fav_col_list',
   },
   {
-    type: 'integer',
+    type: QuestionTypeName.integer,
     $kuid: 'ka9pv41',
     label: ['Favourite number', 'Ulubiona liczba'],
     required: false,
     $autoname: 'Favourite_number',
   },
   {
-    type: 'end_group',
+    type: GroupTypeEndName.end_group,
     $kuid: '/fx8qb06',
   },
 ]
 
-export const simpleSurveyChoices = [
+export const simpleSurveyChoices: SurveyChoice[] = [
   {
     name: 'pink',
     label: ['Pink'],
     list_name: 'fav_col_list',
+    $autovalue: '',
+    $kuid: '',
   },
   {
     name: 'blue',
     label: ['Blue'],
     list_name: 'fav_col_list',
+    $autovalue: '',
+    $kuid: '',
   },
 ]
 
-export const simpleSurveySubmission = {
+export const simpleSurveyAsset: AssetResponse = {
```

Comment:
What do you say to use `as const satisfies Type` for mocks? I think that's more convenient. The `as const` will better reflect the nature of the mock being specific, and `satisfies Type` will assert type-safety whether we update types.

The differences:
```ts
interface Test {
  a: number
  b?: string
}

const v = { a: 1 } satisfies Test
// const v: {
//     a: number;
// }
const v2 = { a: 1 } as const satisfies Test
// const v2: {
//     readonly a: 1;
// }
const v3: Test = { a: 1 }
// const v3: Test
```